wandb/old/settings.py

```diff
@@ -106,9 +107,12 @@ def _settings(default_settings={}):
 
     @staticmethod
     def _global_path():
-        config_dir = os.environ.get(
-            env.CONFIG_DIR, os.path.join(os.path.expanduser("~"), ".config", "wandb")
-        )
+        default_config_dir = os.path.join(os.path.expanduser("~"), ".config", "wandb")
+        # if not writable, fall back to a temp directory
+        if not os.access(default_config_dir, os.W_OK):
```

Comment:
This is false if `~/.config/wandb` doesn't exist. See https://weightsandbiases.slack.com/archives/C01KQ5KTDC3/p1701212657277469?thread_ts=1701095384.672609&cid=C01KQ5KTDC3